### PR TITLE
[WIP] add support for custom delimiter and separator

### PR DESCRIPTION
**Pull request by jipiboily**
Hello folks!

I need to have custom (thousand) delimiter and (fraction) separator to support users from all around the world.

I hacked this version that works for my use case, but I was wondering if you would be interested in having it added to Gruff?

Just FYI, I am not expecting this PR to pass tests or be complete whatsoever.

Let me know your thoughts.

### CODE REVIEW

**Comment by jipiboily** [comment]
@Watson1978 I see you are giving Gruff some love in the past few months, and that's awesome!

I would love get back to the main gem and stop using my fork.

But for that, I would need support for custom delimiter and support like in this PR.

Is that something you are interested in merging in eventually?

Thanks!

**Comment by Watson1978** [comment]
I think It would be nice to have an attribute or methods to customize delimiter.

**Comment by jipiboily** [comment]
Sounds good!

How's the approach I hacked a few months ago, besides the lack of test?

Do you think it's a decent approach or should I look into something different?

Happy to help, but if you can point me roughly to what you have mind, I'll see how / when I can do it.

Thanks!

File: lib/gruff/base.rb [hunk 1]
```diff
@@ -166,7 +166,7 @@ class Base
     #
     # @param target_width [Numeric, String] The graph image width.
     #
-    def initialize(target_width = DEFAULT_TARGET_WIDTH)
+    def initialize(target_width = DEFAULT_TARGET_WIDTH, delimiter: ',', separator: '.')
```
Comment (Watson1978) [inline lib/gruff/base.rb:169]: If you will want to add more arguments, I think they would only settings for the canvas in here.

Comment (Watson1978) [reply lib/gruff/base.rb:169]: Please add comments according to the yard notation if you change something in public API.

Comment (Watson1978) [reply lib/gruff/base.rb:169]: I will add an attribute or a method to set it if needed

**Comment by Watson1978** [comment]
Each version of ImageMagick or each platform produces slightly different images, so you will need to test them using docker on your environment.
Please refer how to run test https://github.com/topfunky/gruff#development

File: lib/gruff/base.rb (hunk 1 above)
Comment (jipiboily) [reply lib/gruff/base.rb:169]: Thanks for the feedback! I'll check it out soon-ish.

**Comment by Watson1978** [comment]
I guess https://github.com/topfunky/gruff/pull/510 is sufficient to make the functionality you want.

**Pull request closed without merging**